### PR TITLE
pybind/mgr: new 'hello world' mgr module skeleton

### PR DESCRIPTION
**Pull request by yaarith**

### CODE REVIEW

**Comment by tchaikov** [comment]
what is the purpose of the hello_world module? is it supposed to be a sample for writing a plugin for mgr?

**Comment by yaarith** [comment]
Indeed, a sample skeleton for a mgr module. Still need to add some examples of how to print to stdout and how to use command args.

**Comment by liewegas** [comment]
Yeah

**Comment by jcsp** [comment]
Hmm, not sure if this needs to be in with the "real" modules if it's purely for documentation purposes?  We do already have the `selftest` module for exercising functionality in tests.

**Comment by liewegas** [comment]
Here is nice because it is less likely to go stale, but we can move it into the docs, too.  Either way!

**Comment by LenzGr** [comment]
I personally would prefer it to be part of the "real" modules, as this makes it easier to find and ensures it does not go stale (like @liewegas  already mentioned). Having such a skeleton is a great way of getting started with a new module.

**Comment by jcsp** [comment]
I'm not deeply committed either way -- however, if we do put it in the main modules path, then let's have some reference to it from the documentation (and possibly a reference from this module's docstring *to* the documentation so that people know there's more?).  If we want to avoid this module going stale then there should also be a test in qa/tasks/mgr/test_module_selftest that enables this module and then checks that no health warnings are raised.

**Comment by jcsp** [comment]
@yaarith are you planning to come back to this?

**Comment by yaarith** [comment]
yes, will be done hopefully today.

**Comment by jcsp** [comment]
Oops, I didn't notice this had been updated (it still had DNM in the title).

**Comment by jcsp** [comment]
retest this please (jenkins)

**Comment by jcsp** [comment]
Targeting for mimic, will add another commit that excludes hello.py from packages after this

**Comment by jcsp** [comment]
docs build failing with (transient?) failure to fetch from git 

retest this please (jenkins)

**Comment by jcsp** [comment]
@yaarith before we merge this, please can you add a link to your `hello.rst` into `doc/mgr/index.rst` along with the other modules, so that the docs get built.

File: src/pybind/mgr/hello/__init__.py [hunk 1]
```diff
@@ -0,0 +1 @@
+from module import *  # NOQA
```
Comment (tchaikov) [inline src/pybind/mgr/hello/__init__.py]: @yaarith might want to change this to
```py
from .module import Module
```

as it's compatible with py3. and is more explicit in the sense that all we need is just the Module class.

File: src/pybind/mgr/hello/module.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
+
+"""
+A hello world module
+
+See doc/mgr/hello.rst for more info.
+"""
+
+from mgr_module import MgrModule
+
+class Module(MgrModule):
```
Comment (tchaikov) [inline src/pybind/mgr/hello/module.py]: nit, could rename this class to Hello to emphasis the fact that any class inheriting from `MgrModule` will be considered as a mgr plugin.

File: src/pybind/mgr/hello/module.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+"""
+A hello world module
+
+See doc/mgr/hello.rst for more info.
+"""
+
+from mgr_module import MgrModule
+
+
+# TODO check why class Hello does not work
```
Comment (tchaikov) [inline src/pybind/mgr/hello/module.py]: will do.

File: src/pybind/mgr/hello/module.py (hunk 2 above)
Comment (yaarith) [reply src/pybind/mgr/hello/module.py]: I tried to rename it to 'Hello', but it did not work; from out/mgr.x.log:
2018-04-25 10:05:11.963 7fdd3dd64700 -1 mgr load_subclass_of Module not found: 'hello'
2018-04-25 10:05:11.963 7fdd3dd64700 -1 mgr load_subclass_of Traceback (most recent call last):
  File "/home/yaarit/ceph/ceph/src/pybind/mgr/hello/__init__.py", line 1, in <module>
    from .module import Module
ImportError: cannot import name Module

2018-04-25 10:05:11.963 7fdd3dd64700 -1 mgr load Class not found in module 'hello'
2018-04-25 10:05:11.963 7fdd3dd64700 -1 mgr init Error loading module 'hello': (2) No such file or directory

What am I missing?

Comment (tchaikov) [reply src/pybind/mgr/hello/module.py]: ahh, you need change
```py
from .module import Module
```
to 
```py
from .module import Hello
```

**Comment by yaarith** [comment]
@jcsp I updated `doc/mgr/index.rst` with `hello` module, following the format of the other modules on the list; but I'm not sure that's what you meant... (didn't see any other .rst links in `doc/mgr/index.rst`) 
Please let me know if I need to change this. Thanks!

**Comment by tchaikov** [comment]
http://pulpito.ceph.com/kchai-2018-04-26_07:48:28-rados-wip-kefu-testing-2018-04-26-1308-distro-basic-smithi/

**Pull request merged**